docs/reference: Update and Extend Slot/System Status File Documentation (#1579)
rauc/rauc: docs/reference.rst

.. _slot-status:

Slot Status File
----------------

For each slot, RAUC tracks some status data like the state of the slot, the
origin of its content and a minimal installation history.
RAUC provides this information via the :ref:`GetSlotStatus
<gdbus-method-de-pengutronix-rauc-Installer.GetSlotStatus>` D-Bus method or
prints it when ``rauc status`` is called with the ``--detailed`` options.

The recommended way to store slot data is on a shared partition within
RAUC's :ref:`data-directory <data-directory>`.
When the data directory is enabled, the file ``<data-directory>/central.raucs``
is used to store information for all slots.
The same file is also used for storing :ref:`system status <system-status>`
information.
If no shared partition is available, RAUC can store the status file as
``/slot.raucs`` on each slot that contains a writable filesystem.

> Slots without a writable filesystem will not have any status data stored then.

Reviewer: ```suggestion
Slots without a writable filesystem will not have any status data stored in this case.
```

Like the configuration files used by RAUC, the slot status files use a
key-value syntax, similar to that found in .ini files.

The information for each slot is grouped in a section like ``[slot.rootfs.1]``
for the central status file, or ``[slot]`` in case of a per-slot file.

.. code-block:: cfg

  [slot.rootfs.0]
  bundle.compatible=FooCorp Super BarBazzer
  bundle.version=2024.08-1
  bundle.description=Introduction of Galactic Feature XYZ
  bundle.build=2024.08.1/imx8mp/20240824-7
  status=ok
  sha256=b14c1457dc10469418b4154fef29a90e1ffb4dddd308bf0f2456d436963ef5b3
  size=419430400
  installed.transaction=dad3289a-7de1-4ad2-931e-fb827edc6496
  installed.timestamp=2024-09-06T09:51:13Z
  installed.count=3

The properties ``bundle.compatible``, ``bundle.version``, ``bundle.description``
and ``bundle.build`` are copies of the respective manifest properties.
More information can be found in this :ref:`subsection <sec-manifest-update>` of
section :ref:`Manifest <sec_ref_manifest>`.

The ``status`` field records the status of each slot.
It can have the following values:

> :ok: The latest updated for this slot succeeded. Its content should be valid.

Reviewer: ```suggestion
:ok: The latest update for this slot succeeded. Its content should be valid.
```

activated or if its activation is still pending.

.. _system-status:

System Status File
------------------

The system status is only available if a central status file is configured for
RAUC (by setting :ref:`data-directory <data-directory>`).

> The system status is managed in the same file as the :ref:`slot status

Reviewer: ```suggestion
The system status is stored in the same file as the :ref:`slot status
```
